internal/api/errors_test.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandleResponseErrorWithHTTPError(t *testing.T) {
+	examples := []struct {
+		HTTPError    *HTTPError
+		APIVersion   string
+		ExpectedBody string
+	}{
+		{
+			HTTPError:    badRequestError(ErrorCodeBadJSON, "Unable to parse JSON"),
+			APIVersion:   "",
+			ExpectedBody: "{\"code\":400,\"error_code\":\"" + ErrorCodeBadJSON + "\",\"msg\":\"Unable to parse JSON\"}",
+		},
+		{
+			HTTPError:    badRequestError(ErrorCodeBadJSON, "Unable to parse JSON"),
+			APIVersion:   "2023-12-31",
+			ExpectedBody: "{\"code\":400,\"error_code\":\"" + ErrorCodeBadJSON + "\",\"msg\":\"Unable to parse JSON\"}",
+		},
+		{
+			HTTPError:    badRequestError(ErrorCodeBadJSON, "Unable to parse JSON"),
+			APIVersion:   "2024-01-01",
+			ExpectedBody: "{\"code\":\"" + ErrorCodeBadJSON + "\",\"message\":\"Unable to parse JSON\"}",
+		},
+		{
+			HTTPError: &HTTPError{
+				HTTPStatus: http.StatusBadRequest,
+				Message:    "Uncoded failure",
+			},
+			APIVersion:   "2024-01-01",
+			ExpectedBody: "{\"code\":\"" + ErrorCodeUnknown + "\",\"message\":\"Uncoded failure\"}",
+		},
+		{
+			HTTPError: &HTTPError{
+				HTTPStatus: http.StatusInternalServerError,
+				Message:    "Unexpected failure",
+			},
+			APIVersion:   "2024-01-01",
+			ExpectedBody: "{\"code\":\"" + ErrorCodeUnexpectedFailure + "\",\"message\":\"Unexpected failure\"}",
+		},
+	}
```

Comment:
Maybe add a bit more cases here for WeakPasswordError and internal server errors.